### ISSUE TITLE
Move content of matrix_tools.hpp, generic_matrix_tools.hpp etc to member functions

### ISSUE THREAD

**Issue by jaeandersson** [casadi/casadi#1285]
The original idea of creating matrix_tools.hpp was to build up a collection of auxiliary tools and to keep the core class clean. In practise, it has proven very difficult to make such a separation. Since the MATLAB interface #176 requires to have them as member functions, it's time to just move them all inside the class and provide a wrapper in its old place.

This should have no effect on user syntax.

**Comment by jaeandersson** [comment]
@jgillis - in #1301:
> Does this mean that the various "tools" files are a thing of the past now?

There are some remnants. Let me follow up when I'm ready to close this issue.

**Comment by jaeandersson** [comment]
Remaining issues in matrix_tools.hpp
  * pinv and solve:  these contain some DMatrix-specific signatures. I think it's ok to have a DMatrix specific implementation (though I doubt that it should it's motivated in this case), but I don't think we should have additional overloads for DMatrix. So the code should be refactored IMO.
  * repmat and sparse: these have names that conflict with Matrix::repmat and Matrix::sparse respectively.

There are also also remaining functions in sx_tools.hpp and mx_tools.hpp. But this should be possible to resolve.

**Comment by jaeandersson** [comment]
Closing this issue. Minor issues remaining:
 * Wrapper functions still exist in mx_tools.hpp and sx_tools.hpp. These should be moved into their respective classes (MX, GenericMatrix or Matrix) and be declared friend. Will postpone this until after #1301 has been resolved to avoid additional code duplication.
 * solve, pinv, evalf should be refactored or removed  #1302
